backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -302,10 +307,12 @@ public Response create(@PathParam("vaultId") String vaultId, VaultDto vaultDto)
 		}
 	}
 
-	public record VaultDto(@JsonProperty("id") String id, @JsonProperty("name") String name, @JsonProperty("description") String description,
+	public record VaultDto(@JsonProperty("id") @NotBlank @Pattern(regexp = ValidationUtil.ID_PATTERN) String id,
+						   @JsonProperty("name") @NotBlank @Pattern(regexp = ValidationUtil.NAME_PATTERN) String name,
+						   @JsonProperty("description") @Pattern(regexp = "(?U)[-\\p{Alnum}\\h_.~!%&'()*+;=:/?#@]*") String description,
```

Comment:
pure madness :D this is too much complexity, which may introduce errors. if we want to allow _some_ legal content,  such as (reading your tests) URLs, please use a different approach:

Create a new annotation class `SafeText`, which itself is annotated with `@Constraint(validatedBy = SafeTextValidator.class)`.  Then add org.owasp.esapi to our dependencies and create the `class SafeTextValidator implements ConstraintValidator<...>` which calls esapi for input validation.

We should then also use DOMPurifier on the frontend, but this is a different PR.